### ISSUE TITLE
[BK-8] Multi-Model Ensemble Strategy

### ISSUE THREAD

**Issue by bbkenny** [X-Aegis/Aegis-backend-ai#12]
### 📝 Description

Implement an ensemble of models to improve prediction robustness.

### 🛠 Tech Stack

• Language: Python (FastAPI)
• Role: AI & Backend Engineering

### ✅ Acceptance Criteria

  - Add ARIMA/GARCH model as a statistical baseline.
  - Add a gradient-boosted model (XGBoost/LightGBM) for feature-rich predictions.
  - Implement weighted ensemble combiner with configurable weights.
  - Track individual model performance to dynamically adjust weights.
  - Create endpoint: `GET /risk/breakdown` showing per-model scores.
  - Write tests for ensemble edge cases (model disagreement scenarios).

---

### ⚠️ Must Do:

- [ ] **Verify Integrity**: Before submitting your PR, ensure all checks pass locally:
  - **Smart contracts**: `cd smartcontract && cargo test --all && cargo build --target wasm32-unknown-unknown --release`
  - **Frontend**: `cd frontend && npm install --legacy-peer-deps && npm run build`
  Your PR will be reviewed, and if either step fails, the PR will not be merged until all checks pass.
- [ ] **Correct Implementation**: Each issue must be solved accurately as it will undergo a thorough review before being merged. It will only be merged if it fully meets the specified requirements and effectively addresses the intended problem.
- [ ] **PR Requirements**: In your Pull Request (PR), you must specify the issue index number and title.
  - **Optional: Visual Proof**: You can optionally provide a screenshot showing the frontend working or the smart contract tests passing/successfully building (e.g., terminal output or UI screenshot).
- [ ] **Link the Issue**: In your PR description, use keywords to automatically close the issue when merged:
  - `Closes #26` — use this when your PR resolves the issue
  - `Fixes #26` — use this when your PR fixes a bug
  - `Resolves #26` — use this when your PR fully resolves it
  Example:
  ```
  ## Summary
  - Added detailed event logging for AI analysis
  
  Closes #26
  ```
- [ ] **Tag Both Issue & PR**: 
  - On the **issue**: Comment tagging the author (`@username`) to let them know you're working on it
  - On the **PR**: Tag `@bbkenny` to notify maintainer(s) review is ready
Example on issue:
    > Hey @bbkenny, I've got a PR ready for this. Want to check it out when you get a chance?
    Example on PR comment:
    > @bbkenny, PR is ready for review!

**Comment by bbkenny** [comment]
quick check @ishitab02 — still nothing pushed on BK-8. I'll need to reassign before the campaign closes if there's no PR. ping me if you're on it.

**Comment by ishitab02** [comment]
hey @bbkenny, thanks for checking in! please go ahead and reassign this to someone else as I'm not currently participating in the campaign. Apologies for the hold-up!
